ownCloud/Client/Actions/Scanner/ScanViewController.swift

```diff
@@ -0,0 +1,526 @@
+ *
+ */
+
+import UIKit
+import VisionKit
+import ImageIO
+import ownCloudSDK
+import ownCloudApp
+
+class ScanImageView : UIImageView {
+	var aspectHeight : CGFloat? {
+		didSet {
+			self.invalidateIntrinsicContentSize()
+		}
+	}
+
+	override var intrinsicContentSize: CGSize {
+		if let aspectHeight = aspectHeight, let image = image {
+			let imageSize = image.size
+			var intrinsicSize : CGSize = CGSize(
+				width: imageSize.width * aspectHeight / imageSize.height,
+				height: aspectHeight
+			)
+
+			if intrinsicSize.width < 1 { intrinsicSize.width = 1 }
+			if intrinsicSize.height < 1 { intrinsicSize.height = 1 }
+
+			return intrinsicSize
+		}
+
+		return super.intrinsicContentSize
+	}
+}
+
+class ScanPage {
+	var image : UIImage
+
+	init(with image: UIImage) {
+		self.image = image
+	}
+}
+
+class ScanPageCell : UICollectionViewCell, Themeable {
+	private var imageView : ScanImageView?
+
+	var page : ScanPage? {
+		didSet {
+			self.imageView?.image = page?.image
+			self.imageView?.invalidateIntrinsicContentSize()
+		}
+	}
+
+	var aspectHeight : CGFloat = 0 {
+		didSet {
+			if aspectHeight != 0 {
+				imageView?.aspectHeight = aspectHeight
+			}
+		}
+	}
+
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+
+		imageView = ScanImageView()
+		imageView?.translatesAutoresizingMaskIntoConstraints = false
+		imageView?.contentMode = .scaleAspectFill
+		imageView?.setContentCompressionResistancePriority(.required, for: .horizontal)
+		imageView?.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+
+		guard let imageView = imageView else { return }
+
+		self.contentView.addSubview(imageView)
+
+		NSLayoutConstraint.activate([
+			imageView.leftAnchor.constraint(equalTo: contentView.leftAnchor),
+			imageView.rightAnchor.constraint(equalTo: contentView.rightAnchor),
+			imageView.topAnchor.constraint(equalTo: contentView.topAnchor),
+			imageView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+		])
+
+		self.layer.shadowColor = UIColor.black.cgColor
+		self.layer.shadowRadius = 6
+		self.layer.shadowOpacity = 0.2
+		self.layer.shadowOffset = CGSize(width: 0, height: 4)
+
+		Theme.shared.register(client: self, applyImmediately: true)
+	}
+
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+	}
+
+	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
+		self.backgroundColor = collection.tableRowColors.backgroundColor
+	}
+}
+
+class ScanPagesCollectionViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout, Themeable {
+	var flowLayout : UICollectionViewFlowLayout
+
+	var pages : [ScanPage] = [] {
+		didSet {
+			self.collectionView.reloadData()
+		}
+	}
+
+	private var height : CGFloat
+	private var verticalPadding : CGFloat = 25
```

Comment:
@felix-schwarz Use `let` for constants instead of `var`

Reply:
Good point! Done.